pvlib/iotools/cams.py

```diff
@@ -0,0 +1,357 @@
+
+    *Parameters only returned if service='cams_radiation'. For description of
+    additional output parameters in verbose mode, see [1]_ and [2]_.
+
+    Note that it is recommended to specify the latitude and longitude to at
+    least the fourth decimal place.
+
+    Variables corresponding to standard pvlib variables are renamed,
+    e.g. `sza` becomes `solar_zenith`. See the
+    `pvlib.iotools.cams.CAMS_RADIATION_VARIABLE_MAP` dict for the complete
+    mapping.
+
+    See Also
+    --------
+    pvlib.iotools.read_cams_radiation, pvlib.iotools.parse_cams_radiation
+
+    Raises
+    ------
+    requests.HTTPError
+        If the request is invalid, then an XML file is returned by the CAMS
+        service and the error message will be raised as an expcetion.
+
+    References
+    ----------
+    .. [1] `CAMS Radiation Service Info
+       <http://www.soda-pro.com/web-services/radiation/cams-radiation-service/info>`_
+    .. [2] `CAMS McClear Service Info
+       <http://www.soda-pro.com/web-services/radiation/cams-mcclear/info>`_
+    .. [3] `CAMS McClear Automatic Access
+       <http://www.soda-pro.com/help/cams-services/cams-mcclear-service/automatic-access>`_
+    """
+    if time_step in TIME_STEPS_MAP.keys():
+        time_step_str = TIME_STEPS_MAP[time_step]
+    else:
+        warnings.warn('Time step not recognized, 1 hour time step used!')
+        time_step, time_step_str = '1h', 'PT01H'
+
+    if (verbose is True) & ((time_step != '1min') | (time_ref != 'UT')):
+        verbose = False
+        warnings.warn("Verbose mode only supports 1 min. UT time series!")
+
+    # Format verbose variable to the required format: {'true', 'false'}
+    verbose = str(verbose).lower()
+
+    if altitude is None:  # Let SoDa get elevation from the NASA SRTM database
+        altitude = -999
+
+    # Start and end date should be in the format: yyyy-mm-dd
+    start_date = start_date.strftime('%Y-%m-%d')
+    end_date = end_date.strftime('%Y-%m-%d')
+
+    email = email.replace('@', '%2540')  # Format email address
+    service = 'get_{}'.format(service.lower())  # Format CAMS service string
+
+    # Manual format the request url, due to uncommon usage of & and ; in url
+    url = ("http://{}/service/wps?Service=WPS&Request=Execute&"
+           "Identifier={}&version=1.0.0&RawDataOutput=irradiation&"
+           "DataInputs=latitude={};longitude={};altitude={};"
+           "date_begin={};date_end={};time_ref={};summarization={};"
+           "username={};verbose={}"
+           ).format(server, service, latitude, longitude, altitude, start_date,
+                    end_date, time_ref, time_step_str, email, verbose)
+
+    res = requests.get(url)
+
+    # Invalid requests returns helpful XML error message
+    if res.headers['Content-Type'] == 'application/xml':
+        errors = res.text.split('ows:ExceptionText')[1][1:-2]
+        raise requests.HTTPError(errors, response=res)
```

Comment:
can we use [`raise_for_status`](https://requests.readthedocs.io/en/master/api/#requests.Response.raise_for_status) instead of the manual error handling here?

Reply:
The API does not return HTTP errors as you would expect from normal services. If a request is invalid, an XML file is returned instead of a csv file and the service fails at throwing any HTTP errors (invalid request return the HTTP status code 200 and res.ok = True).

Line 199 extracts the error message embedded in the XML message.

    errors = res.text.split('ows:ExceptionText')[1][1:-2]

Reply:
Ah, that is unfortunate. Please expand the comment to state that the API still returns HTTP status code 200 so `res.raise_for_status()` will not work.